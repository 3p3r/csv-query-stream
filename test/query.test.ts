import path from "path";
import { query as queryForCoverage } from "../index";
// @ts-ignore
import { query as queryForProduction } from "../dist";

const query =
	process.env.NODE_ENV === "production" ? queryForProduction : queryForCoverage;
const testFile = path.join(__dirname, "costs.zip");

describe("query() tests", () => {
	it("should always pass", () => {
		expect(true).toBe(true);
	});

	it("should return two results (full query)", async () => {
		const results = await query(testFile, {
			quotes: [
				{
					usd: 0.0003,
					unitId: 4,
					termId: 1,
					usageId: 2,
					lengthTagId: 1,
					classTagId: 2,
					optionTagId: 3,
				},
				{
					usd: 1002729,
					unitId: 58,
					termId: 5,
					usageId: 1,
					lengthTagId: 540,
					classTagId: 538,
					optionTagId: 539,
				},
			],
		});
		expect(results).toMatchSnapshot();
	});

	it("should return one result (partial query)", async () => {
		const result = await query(testFile, {
			quotes: [
				{
					usd: 0.1,
					unitId: 11,
					termId: 1,
				},
			],
		});
		expect(result).toMatchSnapshot();
	});

	it("should return two results (partial query - scan)", async () => {
		const result = await query(
			testFile,
			{
				quotes: [
					{
						usd: 0.1,
						unitId: 11,
						termId: 1,
					},
				],
			},
			true,
			{}, // coverage
		);
		expect(result).toMatchSnapshot();
	});

	it("should be able to query by id (line number minus 1)", async () => {
		const result = await query(testFile, {
			serviceTags: [
				{
					id: 360912,
				},
			],
			regions: [
				{
					id: 3,
				},
			],
		});
		expect(result).toMatchSnapshot();
	});
});
