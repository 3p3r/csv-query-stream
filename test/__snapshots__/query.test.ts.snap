// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`query() tests should be able to query by id (line number minus 1) 1`] = `
[
  {
    "id": 360912,
    "serviceId": 53884,
    "tagId": 607,
  },
  {
    "id": 3,
    "name": "eastus",
  },
]
`;

exports[`query() tests should return one result (partial query) 1`] = `
[
  {
    "classTagId": 2,
    "id": 95,
    "lengthTagId": 1,
    "optionTagId": 3,
    "termId": 1,
    "unitId": 11,
    "usageId": 1,
    "usd": 0.1,
  },
]
`;

exports[`query() tests should return two results (full query) 1`] = `
[
  {
    "classTagId": 2,
    "id": 4,
    "lengthTagId": 1,
    "optionTagId": 3,
    "termId": 1,
    "unitId": 4,
    "usageId": 2,
    "usd": 0.0003,
  },
  {
    "classTagId": 538,
    "id": 83061,
    "lengthTagId": 540,
    "optionTagId": 539,
    "termId": 5,
    "unitId": 58,
    "usageId": 1,
    "usd": 1002729,
  },
]
`;

exports[`query() tests should return two results (partial query - scan) 1`] = `
[
  {
    "classTagId": 2,
    "id": 95,
    "lengthTagId": 1,
    "optionTagId": 3,
    "termId": 1,
    "unitId": 11,
    "usageId": 1,
    "usd": 0.1,
  },
  {
    "classTagId": 2,
    "id": 140254,
    "lengthTagId": 1,
    "optionTagId": 6427,
    "termId": 1,
    "unitId": 11,
    "usageId": 1,
    "usd": 0.1,
  },
]
`;
